spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/ThreadPool.scala

```diff
@@ -81,7 +90,8 @@ class ThreadPool(private var poolSize: Int) {
   def setMKLThread(size: Int): this.type = this.synchronized {
     mklPoolSize = Some(size)
     (1 to poolSize).map(i => Future {
-      ThreadPool.setThreadsOfBackend(size)
+      require(MKL.isMKLLoaded)
+      MKL.setNumThreads(size)
```

Comment:
there are two kinds of backends. I think it's better to define two fun setThreadsOfMklBackend and setThreadsOfMkldnnBackend. And different threads call different backends according to their need.

Reply:
Yes. It will be more clear.